lib/appmap/value_inspector.rb

```diff
@@ -32,8 +34,12 @@ def detect_schema(value, max_depth: MAX_DEPTH, type_info: {}, observed_values: S
           detect_schema(next_value, **{ max_depth: max_depth, type_info: value_schema, observed_values: observed_values, depth: depth + 1 })
         end.compact
         type_info[:properties] = properties unless properties.empty?
-      elsif value.respond_to?(:first)
-        detect_schema(value.first, **{ max_depth: max_depth, type_info: type_info, observed_values: observed_values, depth: depth + 1 })
+      elsif value.respond_to?(:map)
+        type_info[:properties] = value.map do |next_value|
```

Comment:
This could generate a large amount of data

Reply:
I've added a limit (5)